src/modules/email/domain/email-generator.ts

```diff
@@ -90,12 +92,13 @@ export class EmailGenerator {
         return removedDiacritics.join('-');
     }
 
-    private async getNextAvailableAddress(calculatedAddress: string): Promise<string> {
-        if (!(await this.emailRepo.existsEmailAddress(calculatedAddress + EmailGenerator.EMAIL_SUFFIX))) {
+    private async getNextAvailableAddress(calculatedAddress: string, emailDomain: string): Promise<string> {
+        if (!(await this.emailRepo.existsEmailAddress(calculatedAddress + '@' + emailDomain))) {
             return calculatedAddress;
         }
         let counter: number = 1;
-        while (await this.emailRepo.existsEmailAddress(calculatedAddress + counter + EmailGenerator.EMAIL_SUFFIX)) {
+        // eslint-disable-next-line no-await-in-loop
+        while (await this.emailRepo.existsEmailAddress(calculatedAddress + counter + '@' + emailDomain)) {
             counter = counter + 1;
         }
```

Comment:
Das await in der Loop könnte man durch eine rekusive Herangehensweise unterbinden. Die Methode ruft sich dafür einfach so lange selber auf bis sie eine freie Mail gefunden hat. Außerdem sollte man ein Abbruchkriterium auf z.B 50 loops oder so setzen um eine unendliche Rekursionsschleife und damit einen StackOverflow Error zu vermeiden.

Reply:
Gute Idee, eigentlich auch ziemlich naheliegend, ist nun rekursiv